pkg/gc/config.go

```diff
@@ -81,3 +82,16 @@ func NewConfigFromConfigMap(configMap *corev1.ConfigMap) (*Config, error) {
 
 	return &c, nil
 }
+
+// SanityCheckedConfigMapLoader provides sanity checking for minRevisionTimeout and returns an error if this value is invalid
+func SanityCheckedConfigMapLoader(minRevisionTimeout time.Duration) func(configMap *corev1.ConfigMap) (*Config, error) {
+	return func(configMap *corev1.ConfigMap) (*Config, error) {
+		cfg, err := NewConfigFromConfigMap(configMap)
+		if cfg != nil && cfg.StaleRevisionTimeout-cfg.StaleRevisionLastpinnedDebounce < minRevisionTimeout {
+			err := fmt.Errorf("Got revision timeout of %v, minimum supported value is %v", cfg.StaleRevisionTimeout, minRevisionTimeout)
+			cfg.StaleRevisionTimeout = minRevisionTimeout
```

Comment:
This line needs to be `minRevisionTimeout + cfg.StaleRevisionLastpinnedDebounce` as well.